core/dbt/context/common.py

```diff
@@ -360,6 +361,15 @@ def generate_config_context(cli_vars):
     return _add_tracking(context)
 
 
+def _load_agate_table(path):
+    try:
+        table = dbt.clients.agate_helper.from_csv(path)
+    except ValueError as e:
+        dbt.exceptions.raise_compiler_error(str(e))
+    table.original_abspath = os.path.abspath(path)
+    return table
```

Comment:
should we be concerned about exposing a context fn that can open arbitrary files?

Reply:
That's a very valid concern. It should probably only let you load from the current context's node - I'll tweak it to be less general.